packages/@aws-cdk/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -285,6 +285,14 @@ export interface FlowLogOptions {
    * @default FlowLogDestinationType.toCloudWatchLogs()
    */
   readonly destination?: FlowLogDestination;
+
+  /**
+   * The fields to include in the flow log record, in the order in which they should appear.
+   * See https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs.html#flow-log-records
+   *
+   * @default - No custom log format options provided.
+   */
+  readonly logFormat?: string;
```

Comment:
Since the log format expects specific values, I think it makes sense to turn this into something that is better typed and can provide useful information to the user. Maybe something like

```ts
readonly customLogFormatFields?: LogFormatField[];
```

And then `LogFormat` could be something like

```ts
export class LogFormatField {
  /**
    * The source port of the traffic.
    */
  public static readonly SRC_PORT = new LogFormatField('${srcport'});

  public static custom(field: string): LogFormatField {
    return new LogFormatField(`\${${field}}`);
  }

  constructor(public readonly value: string) {}

}

new FlowLog(stack, 'FlowLogs', {
  customLogFormatFields: [
    LogFormatField.SRC_PORT,
    LogFormatField.custom('dstport'),
    ...
  ]
}
```

Reply:
Good feedback, will work on it